File: atcoder/abc137B_one_clue.hs
-- unihernandez22
-- https://atcoder.jp/contests/abc137/tasks/abc137_b
-- math

import Data.List (intercalate)

main = do
  li <- getLine
  let k:x:[] = map read $ words li
  putStrLn $ intercalate " " $ map show $
    if x - k + 1 < -10^6 then
      [-10^6 .. x+k-1]
    else if x + k - 1 > 10^6 then
      [10^6 .. x-k+1]
    else
      [x-k+1 .. x+k-1]

File: atcoder/abc137B_one_clue.ml
(*(* vicfred
 * https://atcoder.jp/contests/abc137/tasks/abc137_b
 * implementation
 * *)

let rec solve i n =
    if i > n then Printf.printf "\n" else
        (Printf.printf "%d " i; solve (i+1) n);;

Scanf.scanf "%d %d" @@ fun k x ->
    solve (x - k + 1) (x + k - 1);;

